src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -318,10 +322,38 @@ object TaskBuilder {
         "MESOS_TASK_ID" -> taskId.map(_.getValue),
         "MARATHON_APP_ID" -> Some(app.id.toString),
         "MARATHON_APP_VERSION" -> Some(app.version.toString),
-        "MARATHON_APP_DOCKER_IMAGE" -> app.container.flatMap(_.docker.map(_.image))
+        "MARATHON_APP_DOCKER_IMAGE" -> app.container.flatMap(_.docker.map(_.image)),
+        "MARATHON_APP_RESOURCE_CPUS" -> Some(app.cpus.toString),
+        "MARATHON_APP_RESOURCE_MEM" -> Some(app.mem.toString),
+        "MARATHON_APP_RESOURCE_DISK" -> Some(app.disk.toString)
       ).collect {
           case (key, Some(value)) => key -> value
-        }.toMap
+        }.toMap ++ labelsToEnvVars(app.labels)
     }
   }
+
+  def labelsToEnvVars(labels: Map[String, String]): Map[String, String] = {
+    var variables = mutable.Map(
+      "MARATHON_APP_LABELS" -> labels.keys
+        .map(escapeEnvironmentVariableName)
+        .filter(_.length <= maxVariableLength)
+        .mkString(" ")
+    )
+
+    labels.foreach {
+      case (key, value) => {
```

Comment:
You can check the length of key and value here:
`case (key, value) if key.length<512 && value.length<512 => ...`